prediction_market_agent/agents/microchain_agent/functions.py

```diff
@@ -101,19 +105,31 @@ def __call__(self, market_id: str) -> list[str]:
 class PredictPropabilityForQuestion(MarketFunction):
     @property
     def description(self) -> str:
-        return "Use this function to research the probability of an event occuring"
+        return (
+            "Use this function to research perform research and predict the "
+            "probability of an event occuring. Returns the probability. The "
+            "one parameter is the market id of the prediction market you want "
+            "to predict the probability of. Note, this costs money to run."
+        )
 
     @property
     def example_args(self) -> list[str]:
-        return ["Will Joe Biden get reelected in 2024?"]
-
-    def __call__(self, a: str) -> float:
-        if a == "Will Joe Biden get reelected in 2024?":
-            return 0.41
-        if a == "Will Bitcoin hit 100k in 2024?":
-            return 0.22
+        return [get_example_market_id(self.market_type)]
 
-        return 0.0
+    def __call__(self, question: str) -> float:
+        private_key = APIKeys().bet_from_private_key.get_secret_value()
+        with saved_str_to_tmpfile(private_key) as tmpfile_path:
+            response = interact(
```

Comment:
This costs money to run. Should we do some kind of balance check here to see if the agent has enough funds? Otherwise this will throw an error.

Reply:
Added. I don't think there's a way to query the cost of a mech call, so I've hard-coded it here. 

I tried playing around with the value [here](https://github.com/valory-xyz/mech-client/blob/main/mech_client/interact.py#L338) that sets the cost of the mech call, and it fails for anything <0.01xDai, and works for anything >=.

![Uploading Screenshot 2024-04-15 at 11.37.29.png…]()